File: solutions/javascript/bird-watcher/1/bird-watcher.js
/**
 * Calculates the total number of birds counted.
 *
 * @param {number[]} birdsPerDay An array containing the bird count for each day.
 * @returns {number} The total number of birds counted.
 */
export function totalBirdCount(birdsPerDay) {
  let total = 0;
  for (let i = 0; i < birdsPerDay.length; i++) {
    total += birdsPerDay[i];
  }
  return total;
}

/**
 * Calculates the total number of birds counted in a specific week.
 *
 * @param {number[]} birdsPerDay An array containing the bird count for each day.
 * @param {number} week The week number (1-indexed).
 * @returns {number} The total number of birds counted in that week.
 */
export function birdsInWeek(birdsPerDay, week) {
  let weeklyTotal = 0;
  const startIndex = (week - 1) * 7;
  const endIndex = startIndex + 7;

  for (let i = startIndex; i < endIndex; i++) {
    if (i < birdsPerDay.length) { // Defensive check
      weeklyTotal += birdsPerDay[i];
    }
  }
  return weeklyTotal;
}

/**
 * Corrects a counting mistake by incrementing the bird count on every second day, starting from the first day.
 *
 * @param {number[]} birdsPerDay An array containing the bird count for each day.
 * @returns {number[]} The corrected array of bird counts.
 */
export function fixBirdCountLog(birdsPerDay) {
  for (let i = 0; i < birdsPerDay.length; i += 2) {
    birdsPerDay[i]++;
  }
  return birdsPerDay;
}

File: solutions/go/bird-watcher/1/bird_watcher.go
package birdwatcher

// TotalBirdCount return the total bird count by summing
// the individual day's counts.
func TotalBirdCount(birdsPerDay []int) int {
	total := 0
	for _, count := range birdsPerDay {
		total += count
	}
	return total
}

// BirdsInWeek returns the total bird count by summing
// only the items belonging to the given week.
func BirdsInWeek(birdsPerDay []int, week int) int {
	// Calculate the start index for the given week (0-based)
	startIndex := (week - 1) * 7
	endIndex := startIndex + 7
	
	// Sum the bird counts for the specified week
	total := 0
	for i := startIndex; i < endIndex && i < len(birdsPerDay); i++ {
		total += birdsPerDay[i]
	}
	return total
}

// FixBirdCountLog returns the bird counts after correcting
// the bird counts for alternate days.
func FixBirdCountLog(birdsPerDay []int) []int {
	// The bird was present on the first day (index 0) and every second day
	// So we need to add 1 to every even index (0, 2, 4, 6, ...)
	for i := 0; i < len(birdsPerDay); i += 2 {
		birdsPerDay[i]++
	}
	return birdsPerDay
}